<?xml version="1.0"?>
<launch>
    <node pkg="twist_controller" type="dbw_node.py" name="dbw_node">
        <param name="vehicle_mass" value="1736.35" />
        <param name="fuel_capacity" value="13.5" />
        <param name="brake_deadband" value=".1" />
        <param name="decel_limit" value="-1." />
        <param name="accel_limit" value="1." />
        <param name="wheel_radius" value="0.2413" />
        <param name="wheel_base" value="2.8498" />
        <param name="steer_ratio" value="14.8" />
        <param name="max_lat_accel" value="3." />
        <param name="max_steer_angle" value="8." />
        <param name="loop_rate" value="30." />
        <param name="is_sim" value="false" />
    </node>

    <node name="throttle_pid" pkg="pid" type="controller">
      <param name="Kp" value="1.0" />
      <param name="Ki" value="0.02" />
      <param name="Kd" value="0.0" />
      <param name="upper_limit" value="0.035" />
      <param name="lower_limit" value="-1.0" />
      <param name="windup_limit" value="20" />
      <param name="pid_enable_topic" value="/throttle_pid/enable" />
      <param name="topic_from_plant" value="/throttle_pid/state" />
      <param name="setpoint_topic" value="/throttle_pid/setpoint" />
      <param name="topic_from_controller" value="/throttle_pid/control_effort" />
      <param name="max_loop_frequency" value="30.0" />
      <param name="min_loop_frequency" value="30.0" />
     </node>

    <!-- enable for debugging PID controller
    <node name="setpoint_node" pkg="pid" type="setpoint_node" output="screen" />
    <node name="rqtplot" pkg="rqt_plot" type="rqt_plot" />
    -->
</launch>
